src/solver/hydro/daily/h2o_j_free.cpp

```diff
@@ -51,7 +51,8 @@ void H2O_J_Free(DONNEES_MENSUELLES* DonneesMensuelles)
             SPX_LibererProbleme(ProbSpx);
     }
 
-    free(ProblemeHydraulique.Probleme);
+    PROBLEME_SIMPLEXE* pb = (PROBLEME_SIMPLEXE*)ProblemeHydraulique.Probleme;
+    delete pb;
```

Comment:
Why do we need to cast here ?

If you really need to cast, please use `static_cast` instead, it offers more guarantees.

Reply:
can't delete a void*, I'll use static cast then

Reply:
Then why do you need to `delete` ?